src/qibolab/sweeper.py

```diff
@@ -1,20 +1,65 @@
 from dataclasses import dataclass
+from enum import Enum, auto
 from typing import Optional
 
 import numpy.typing as npt
 
+from qibolab.pulses import PulseType
+
+
+class Parameter(Enum):
+    """Sweeping parameters."""
+
+    attenuation = auto()
+    gain = auto()
+    frequency = auto()
+    amplitude = auto()
+    current = auto()
+
 
 @dataclass
 class Sweeper:
-    parameter: str
+    """Data structure for Sweeper object.
+
+    This object is passed as an argument to the method :func:`qibolab.platforms.abstract.AbstractPlatform.sweep`
+    which enables the user to sweep a specific parameter for one or more pulses. For information on how to
+    perform sweeps see :func:`qibolab.platforms.abstract.AbstractPlatform.sweep`.
+
+    Example:
+        .. testcode::
+
+            import numpy as np
+            from qibolab.platform import Platform
+            from qibolab.sweeper import Sweeper, Parameter
+            from qibolab.pulses import PulseSequence
+
+
+            platform = Platform("dummy")
+            sequence = PulseSequence()
+            parameter = Parameter.frequency
+            pulse = platform.create_qubit_readout_pulse(qubit=0, start=0)
+            sequence.add(pulse)
+            parameter_range = np.random.randint(10, size=10)
+            sweeper = Sweeper(parameter, parameter_range, [pulse])
+            platform.sweep(sequence, sweeper)
+
+    Args:
+        parameter (`qibolab.sweeper.Parameter`): parameter to be swept, possible choices are frequency, attenuation, amplitude, current and gain.
+        values (np.ndarray): sweep range. If the parameter is `frequency` the sweep will be a shift around the readout frequency
+            in case of a `ReadoutPulse` or around the drive frequency for a generic `Pulse`. For other parameters the sweep will be
+            performed directly over the range specified.
+        pulses (list) : list of `qibolab.pulses.Pulse` to be swept (optional).
+        qubits (lilst): list of `qibolab.platforms.abstract.Qubit` to be swept (optional).
+    """
+
+    parameter: Parameter
     values: npt.NDArray
     pulses: Optional[list] = None
     qubits: Optional[list] = None
 
     @property
-    def pulse_type(self):
-        if self.pulses is not None:
-            pulse_type = self.pulses[0].type.name.lower()
-            for pulse in self.pulses:
-                assert pulse.type.name.lower() == pulse_type
-            return pulse_type
+    def pulse_type(self) -> Optional[PulseType]:
+        types = {p.type for p in self.pulses}
+        if len(types) > 1:
+            raise RuntimeError("Not homogeneous pulses")
+        return next(iter(types), None)
```

Comment:
Are you using this property anywhere in the sweepers you implemented? I know I was using it for QM, but I managed to remove it after updating to use the `Parameter` object. If you are also not using it maybe we can remove it completely to simplify the code, but I can do this once I fully fix the QM part.